core/trino-main/src/test/java/io/trino/sql/query/TestSubqueries.java

```diff
@@ -913,4 +906,399 @@ public void testCorrelatedLeftUnnestWithGlobalAggregation()
                         "ON TRUE "))
                 .matches("VALUES ARRAY[1, 2, 3]");
     }
+
+    @Test
+    public void testCorrelatedUnnestInScalarSubquery()
+    {
+        // in the following queries, the part `SELECT * FROM UNNEST(...)` creates an UnnestNode of type INNER,
+        // while `(VALUES 1) LEFT JOIN UNNEST(...) ON TRUE` creates an UnnestNode of type LEFT
+        assertThat(assertions.query(
+                "SELECT (SELECT * FROM UNNEST(a) u(x)) " +
+                        "FROM (VALUES ARRAY[1], ARRAY[2]) t(a)"))
+                .matches("VALUES 1, 2");
+
+        assertThat(assertions.query(
+                "SELECT (SELECT x FROM (VALUES 1) LEFT JOIN UNNEST(a) u(x) ON TRUE)" +
+                        "FROM (VALUES ARRAY[1], ARRAY[2]) t(a)"))
+                .matches("VALUES 1, 2");
+
+        assertThatThrownBy(() -> assertions.query(
+                "SELECT (SELECT * FROM UNNEST(a) u(x)) " +
+                        "FROM (VALUES ARRAY[1, 2, 3]) t(a)"))
+                .hasMessage("Scalar sub-query has returned multiple rows");
+
+        // limit in subquery
```

Comment:
For limit (without sorting) any value from an array can be returned. So we are just lucky that tests are not flaky. Right?

Reply:
Right. I'll change them so that the arrays have all the same values. It will then still show that LIMIT was applied.